lib/Parse/ParseDecl.cpp

```diff
@@ -5778,7 +5778,8 @@ ParserStatus Parser::parseLineDirective(bool isLine) {
         diagnose(Tok, diag::expected_line_directive_number);
         return makeParserError();
       }
-      if (Tok.getText().getAsInteger(0, StartLine)) {
+      auto text = copyAndStripUnderscores(Tok.getText());
```

Comment:
Could you limit this to only strings containing `_`?
```suggestion
      auto text = Tok.getText();
      if (text.contains('_'))
        text = copyAndStripUnderscores(text);
```
Because `copyAndStripUnderscores` unconditionally copies the string to `ASTContext`, which is unnecessary for this case.

Reply:
Makes sense, just fixed